src/main/java/art/heredium/service/CouponUsageService.java

```diff
@@ -82,4 +84,31 @@ public List<CouponUsage> distributeCoupons(
         });
     return this.couponUsageRepository.saveAll(couponUsages);
   }
+
+  @Transactional(rollbackFor = Exception.class)
+  public void checkoutCouponUsage(String uuid) {
+    CouponUsage couponUsage =
+        couponUsageRepository
+            .findByUuid(uuid)
+            .orElseThrow(() -> new ApiException(ErrorCode.COUPON_NOT_FOUND, "Coupon not found"));
+
+    LocalDateTime now = LocalDateTime.now();
+
+    if (couponUsage.getExpirationDate().isBefore(now)) {
+      throw new ApiException(ErrorCode.COUPON_EXPIRED, "Coupon is expired");
+    }
+
+    if (couponUsage.getIsUsed() && !couponUsage.isPermanent()) {
+      throw new ApiException(ErrorCode.COUPON_ALREADY_USED, "Coupon is already used");
+    }
+
+    if (couponUsage.isPermanent()) {
+      couponUsage.setUsedCount(couponUsage.getUsedCount() + 1);
+    } else {
+      couponUsage.setIsUsed(true);
+    }
```

Comment:
```suggestion
    couponUsage.setUsedCount(couponUsage.getUsedCount() + 1);
    couponUsage.setIsUsed(true);
```
- Luôn tăng usedCount lên 1 trong mọi trường hợp.
- Luôn set isUsed = true

Reply:
I updated in https://github.com/Incuvers-IT/heredium-BE/pull/31/commits/1e926d00e58329d159ba7b86f9d1e1364016a6a9